database/scripts/seed.js

```diff
@@ -60,26 +60,68 @@ const endProgress = () => {
   process.stdout.write(`Progress: 100%`.padEnd(20) + `[${'='.repeat(20)}]`);
 }
 
-const seedDB = async (n = 10000000, uuidBufSiz = 1000) => {
-  if (n > uuidBufSiz) {
-    n /= uuidBufSiz;
+class StatusBar {
+  constructor(limit) {
+    this.limit = limit;
+    console.time(' runtime');
+  }
+  resetCrg() {
+    process.stdout.clearLine();
+    process.stdout.cursorTo(0);
+  }
+  writeProgress(prog) {
+    this.resetCrg();
+    let tenth = (prog / this.limit * 10);
+    let progBar = ('[' + '='.repeat(tenth * 2) + '>'.repeat(tenth < 10)).padEnd(21) + ']';
+    process.stdout.write(`Progress: ${(tenth * 10).toFixed(2)}%`.padEnd(20) + progBar);
   }
 
-  for (let i = 0; i < n; i++) {
-    writeProgress(i, n);
-    // every review needs 2 UUIDs
-    let uuids = mk_nUUIDS(uuidBufSiz << 1);
-    let uptr = 0;
+  endProgress() {
+    this.resetCrg();
+    process.stdout.write(`Progress: 100%`.padEnd(20) + `[${'='.repeat(20)}]`);
+    console.timeEnd(' runtime')
+  }
+}
 
-    let reviews = new Array(uuidBufSiz);
-    for (let j = 0; j < uuidBufSiz; j++) {
-      reviews[j] = genReviewArr(uuids[uptr++], uuids[uptr++]);
+function* uuidFactory(n = 2000) {
+  while (true) {
+    let arr = mk_nUUIDS(n);
+    for (let uuid of arr) {
+      yield uuid;
     }
-    await Promise.all(reviews.map(insertReview));
   }
+}
+
+const seedDB = async (inserter, n = 10000000, bufSize = 1000) => {
+
+  let factory = uuidFactory();
+  let arr = new Array(bufSize);
+  let aptr = 0;
+
+  let status = new StatusBar(n);
+
+  for (let i = 0; i < n; i++) {
+    status.writeProgress(i);
+    let lim = ~~(Math.random() * 24);
+    let prodId = factory.next().value;
+    for (let j = 0; j < lim; j++) {
+      let itemId = factory.next().value;
+      arr[aptr++] = genReviewArr(itemId, prodId);
+      if (aptr >= bufSize) {
+        await Promise.all(arr.map(inserter));
+        aptr = 0;
+      }
+    }
+  }
+
+  status.endProgress();
+}
 
-  // endProgress();
-  console.timeEnd(' runtime');
+//todo: implement marsaglia polar method for generating normdist of product id's
+if (require.main === module) {
+  seedDB(insertReview, 10000000, 1000);
 }
 
-seedDB(10000000, 1000);
\ No newline at end of file
+module.exports = {
+  StatusBar,
+}
```

Comment:
Not a fan of the comma before value syntax but if you're going to use it keep it consistent